### PR TITLE
chore: Copy http2_util, logger_conf and logger_util to util project

### PR DESCRIPTION
**Pull request by ghislainbourgeois**
Copy `http2_util`, `logger_conf` and `logger_util` to `util` project.

Once merged and released, dependent projects will be modified to remove the dependency on the previous small projects to simplify maintenance.

### CODE REVIEW

File: http2_util/server.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
+// Copyright 2019 Communication Service/Software Laboratory, National Chiao Tung University (free5gc.org)
+//
+// SPDX-License-Identifier: Apache-2.0
+
+//go:build !debug
+// +build !debug
+
```
Comment (gab-arrobo) [inline http2_util/server.go:7]:
```suggestion
```

File: http2_util/server_debug.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
+// Copyright 2019 Communication Service/Software Laboratory, National Chiao Tung University (free5gc.org)
+//
+// SPDX-License-Identifier: Apache-2.0
+
+//+build debug
+
```
Comment (gab-arrobo) [inline http2_util/server_debug.go:6]:
```suggestion
```

**Review by gab-arrobo — COMMENTED** [review]
I added two minor edits/comments

File: http2_util/server_debug.go (hunk 2 above)
Comment (ghislainbourgeois) [reply http2_util/server_debug.go:6]: Those comments affect the build process, if I remove them, both files will be built simultaneously and will conflict.

I can simply remove the debug version if we do not require it and then remove the comments from the `server.go` file.

WDYT?

Comment (gab-arrobo) [reply http2_util/server_debug.go:6]: Understood. Then, let's just leave them there. Thanks for the clarification!

**Comment by gab-arrobo** [comment]
@ghislainbourgeois,
How different is this `logger_util/log_level.go` compared to the `logger/logger_conf.go`. To me, they look pretty similar. It looks like `logger` (content) is pretty much the same as `logger_util`, correct?

**Comment by gab-arrobo** [comment]
What about checking whether we can also add the `util_3gpp` into the `util` repo?

**Comment by ghislainbourgeois** [comment]
> @ghislainbourgeois, How different is this `logger_util/log_level.go` compared to the `logger/logger_conf.go`. To me, they look pretty similar. It looks like `logger` (content) is pretty much the same as `logger_util`, correct?

They look quite similar, I will try to make sure we can use only `logger` and get rid of `logger_util` completely.

File: logger_conf/conf.go
Comment (gab-arrobo) [inline logger_conf/conf.go:1]: @ghislainbourgeois,
Does not `logger/logger.go` already have similar content as this file? or am I overlooking something?

**Comment by gab-arrobo** [comment]
@ghislainbourgeois, is this PR ready to merge? Or are you planning to address my previous comment about the `logger/logger/go` file/code?

**Comment by ghislainbourgeois** [comment]
> @ghislainbourgeois, is this PR ready to merge? Or are you planning to address my previous comment about the `logger/logger.go` file/code?

I have started looking in that previous comment, and evaluating if we can easily merge that code. I will let you know today what direction I choose.

File: logger_conf/conf.go
Comment (ghislainbourgeois) [reply logger_conf/conf.go:1]: The files seem to be used in slightly different ways for different loggers. I am not comfortable moving this right now, and I think we should plan a logging cleanup as a separate task after we centralize those dependencies.

**Comment by ghislainbourgeois** [comment]
I think this is ready to merge now, and in the near future we should do a logging cleanup.

**Comment by gab-arrobo** [comment]
> I think this is ready to merge now, and in the near future we should do a logging cleanup.

Ok, thanks. As soon as I merge it, I am going to create a new release/tag

**Review by gab-arrobo — APPROVED** [review]
+1

**Comment by gab-arrobo** [comment]
@ghislainbourgeois,
I just created release/tag [v1.1.0](https://github.com/omec-project/util/releases/tag/v1.1.0) such that we can use it in the other repos. Please free to open PRs in the other repos to replace https2_util, logger_conf, logger_util and util_3gpp repo with this `util v1.1.0` version

**Pull request merged**